src/Parser/Druid/Restoration/Modules/Talents/TreeOfLife.js

```diff
@@ -35,205 +44,146 @@ class TreeOfLife extends Analyzer {
+      } else {
+        this.completedCsUptime += buffUptime;
+      }
 
-  get rejuvenationIncreasedThroughput() {
-    return (this.totalHealingFromRejuvenationDuringToL / 1.15) - (this.totalHealingFromRejuvenationDuringToL / (1.15 * 1.5)); // TODO what's up with this calc?
-  }
-  get rejuvenationIncreasedThroughputPercent() {
-    return this.owner.getPercentageOfTotalHealingDone(this.rejuvenationIncreasedThroughput);
+      this.lastTolCast = null;
+      this.lastTolApply = null;
+    }
   }
 
-  get tolIncreasedThroughput() {
-    return this.totalHealingDuringToL - this.totalHealingDuringToL / 1.15;
+  get hardcastUptime() {
+    const currentUptime = !(this.lastTolCast) ? 0 : Math.min(TOL_DURATION, this.owner.currentTimestamp - this.lastTolCast);
+    return currentUptime + this.completedTolUptime;
   }
-  get tolIncreasedThroughputPercent() {
-    return this.owner.getPercentageOfTotalHealingDone(this.tolIncreasedThroughput);
+  get hardcastUptimePercent() {
+    return this.hardcastUptime / this.owner.fightDuration;
   }
 
-  get rejuvManaThroughput() {
-    return (((this.totalRejuvenationsDuringToL * 10) * 0.3) / 10) * this.rejuvenation.avgRejuvHealing; // TODO what's up with this calc?
+  get csUptime() {
+    let currentUptime = 0;
+    if (this.lastTolApply) {
+      currentUptime = this.owner.currentTimestamp - this.lastTolApply;
+      if (this.lastTolCast) {
+        currentUptime -= Math.min(TOL_DURATION, this.owner.currentTimestamp - this.lastTolApply);
+      }
+    }
+    return currentUptime + this.completedCsUptime;
   }
-  get rejuvManaThroughputPercent() {
-    return this.owner.getPercentageOfTotalHealingDone(this.rejuvManaThroughput);
+  get csUptimePercent() {
+    return this.csUptime / this.owner.fightDuration;
   }
 
-  get wildGrowthIncreasedThroughput() {
-    return this.totalHealingFromWildgrowthsDuringToL / 1.15 - this.totalHealingFromWildgrowthsDuringToL / (1.15 * (8 / 6)); // TODO what's up with this calc?
-  }
-  get wildGrowthIncreasedThroughputPercent() {
-    return this.owner.getPercentageOfTotalHealingDone(this.wildGrowthIncreasedThroughput);
+  _getManaSavedHealing(accumulator) {
+    return accumulator.rejuvManaSaved * this.rejuvenation.avgRejuvHealing;
   }
 
-  get totalThroughput() {
-    return this.rejuvenationIncreasedThroughput + this.tolIncreasedThroughput + this.rejuvManaThroughput + this.wildGrowthIncreasedThroughput;
-  }
-  get totalThroughputPercent() {
-    return this.owner.getPercentageOfTotalHealingDone(this.totalThroughput);
+  _getTotalHealing(accumulator) {
+    return accumulator.allBoostHealing + accumulator.rejuvBoostHealing + accumulator.extraWgHealing + this._getManaSavedHealing(accumulator);
   }
 
   get suggestionThresholds() {
     return {
-      actual: this.totalThroughputPercent,
+      actual: this.owner.getPercentageOfTotalHealingDone(this._getTotalHealing(this.hardcast)),
       isLessThan: {
         minor: 0.11,
```

Comment:
I think we can lower the thresholds, this was more a thing in nighthold where cultivation used to easily do over 11%. I'd say minor : 0.08, average 0.06 and major 0.04 would be appropriate.

Reply:
yeah you're right, worth rethinking for sure

Reply:
Ended up doing 0.09, 0.06, 0.03, because I'm doing a thing where "minor issue is anything other than really good", "major issue is really bad".